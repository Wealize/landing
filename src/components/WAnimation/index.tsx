import React from 'react'
import ReactPlayer from 'react-player'

import useWindowSize from '../../hooks/useWindowSize'
import { BREAKPOINT_MD, BREAKPOINT_LG } from '../../theme/media'

import {
  Video
} from './styles'

const WAnimation = () => {
  const W_768_MP4_VIDEO_URL = '/video/w_768.mp4'
  const W_1400_MP4_VIDEO_URL = '/video/w_1400.mp4'
  const W_2048_MP4_VIDEO_URL = '/video/w_2048.mp4'
  const { width } = useWindowSize()

  const getVideoUrl = () => {
    if (!width || width <= BREAKPOINT_MD) return W_768_MP4_VIDEO_URL

    if (width > BREAKPOINT_MD && width <= BREAKPOINT_LG) return W_1400_MP4_VIDEO_URL

    if (width > BREAKPOINT_LG) return W_2048_MP4_VIDEO_URL
  }

  return (
    <Video className="w-animation">
      <ReactPlayer className="react-player" url={[
        { src: `${getVideoUrl()}`, type: 'video/mp4' }
      ]} width='100%' height='100%' playing autoPlay muted loop stopOnUnmount>
      </ReactPlayer>
    </Video>
  )
}


export default React.memo(WAnimation)
