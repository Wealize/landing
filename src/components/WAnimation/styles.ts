import styled from 'styled-components'

import { media } from '../../theme/media'

export const Video = styled.div<{isMixMultiplyBlendMode: boolean}>`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  > div {
    width: 100%;
    height: 0px !important;
    position: relative;
    padding-top: ${({ isMixMultiplyBlendMode }) => (isMixMultiplyBlendMode ? '100%' : '75%')};

    ${({ isMixMultiplyBlendMode }) => media.greaterThan('md')`
      padding-top: ${isMixMultiplyBlendMode ? '100%' : '75%'};
    `};
  }


  > div video {
    width: 100%;
    height: auto;
    position: absolute;
    top: 0;
    left: 0;
    object-fit: fill;
    z-index: ${({ isMixMultiplyBlendMode }) => (isMixMultiplyBlendMode ? -1 : 1)};
  }
`
