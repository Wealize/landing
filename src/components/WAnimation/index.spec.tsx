import React from 'react'
import { mount } from 'enzyme'

import WAnimation from './index'

describe('<WAnimation />', () => {
  const wrapper = mount(<WAnimation />)

  it('wrapper: should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render content correctly', () => {
    expect(wrapper.find('video').length).toEqual(1)
    expect(wrapper.find('source').length).toEqual(1)
  })
})
