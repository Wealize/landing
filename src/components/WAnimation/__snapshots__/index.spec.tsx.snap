// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WAnimation /> wrapperWithBlendMode: should render without blend mode 1`] = `
<Memo(WAnimation)
  isMixMultiplyBlendMode={true}
>
  <styles__Video
    className="w-animation"
    isMixMultiplyBlendMode={true}
  >
    <div
      className="styles__Video-sc-1iarrbt-0 kRyIUq w-animation"
    >
      <ReactPlayer
        autoPlay={true}
        className="react-player"
        config={
          Object {
            "dailymotion": Object {
              "params": Object {
                "api": 1,
                "endscreen-enable": false,
              },
            },
            "facebook": Object {
              "appId": "1309697205772819",
              "attributes": Object {},
              "playerId": null,
              "version": "v3.3",
            },
            "file": Object {
              "attributes": Object {},
              "dashVersion": "3.1.3",
              "flvVersion": "1.5.0",
              "forceAudio": false,
              "forceDASH": false,
              "forceFLV": false,
              "forceHLS": false,
              "forceVideo": false,
              "hlsOptions": Object {},
              "hlsVersion": "0.14.16",
              "tracks": Array [],
            },
            "mixcloud": Object {
              "options": Object {
                "hide_cover": 1,
              },
            },
            "soundcloud": Object {
              "options": Object {
                "buying": false,
                "download": false,
                "liking": false,
                "sharing": false,
                "show_comments": false,
                "show_playcount": false,
                "visual": true,
              },
            },
            "twitch": Object {
              "options": Object {},
              "playerId": null,
            },
            "vidyard": Object {
              "options": Object {},
            },
            "vimeo": Object {
              "playerOptions": Object {
                "autopause": false,
                "byline": false,
                "portrait": false,
                "title": false,
              },
            },
            "wistia": Object {
              "options": Object {},
              "playerId": null,
            },
            "youtube": Object {
              "embedOptions": Object {},
              "onUnstarted": [Function],
              "playerVars": Object {
                "iv_load_policy": 3,
                "modestbranding": 1,
                "playsinline": 1,
                "rel": 0,
                "showinfo": 0,
              },
            },
          }
        }
        controls={false}
        height="100%"
        light={false}
        loop={true}
        muted={true}
        onBuffer={[Function]}
        onBufferEnd={[Function]}
        onDisablePIP={[Function]}
        onDuration={[Function]}
        onEnablePIP={[Function]}
        onEnded={[Function]}
        onError={[Function]}
        onPause={[Function]}
        onPlay={[Function]}
        onProgress={[Function]}
        onReady={[Function]}
        onSeek={[Function]}
        onStart={[Function]}
        pip={false}
        playbackRate={1}
        playing={true}
        playsinline={false}
        progressInterval={1000}
        stopOnUnmount={true}
        style={Object {}}
        url={
          Array [
            Object {
              "src": "/video/w_1300_blend_mode_multiply.mp4",
              "type": "video/mp4",
            },
          ]
        }
        volume={null}
        width="100%"
        wrapper="div"
      >
        <div
          autoPlay={true}
          className="react-player"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <Suspense
            fallback={null}
          >
            <Player
              activePlayer={
                Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": [Function],
                  "_status": 1,
                }
              }
              autoPlay={true}
              className="react-player"
              config={
                Object {
                  "attributes": Object {},
                  "dailymotion": Object {
                    "params": Object {
                      "api": 1,
                      "endscreen-enable": false,
                    },
                  },
                  "dashVersion": "3.1.3",
                  "facebook": Object {
                    "appId": "1309697205772819",
                    "attributes": Object {},
                    "playerId": null,
                    "version": "v3.3",
                  },
                  "file": Object {
                    "attributes": Object {},
                    "dashVersion": "3.1.3",
                    "flvVersion": "1.5.0",
                    "forceAudio": false,
                    "forceDASH": false,
                    "forceFLV": false,
                    "forceHLS": false,
                    "forceVideo": false,
                    "hlsOptions": Object {},
                    "hlsVersion": "0.14.16",
                    "tracks": Array [],
                  },
                  "flvVersion": "1.5.0",
                  "forceAudio": false,
                  "forceDASH": false,
                  "forceFLV": false,
                  "forceHLS": false,
                  "forceVideo": false,
                  "hlsOptions": Object {},
                  "hlsVersion": "0.14.16",
                  "mixcloud": Object {
                    "options": Object {
                      "hide_cover": 1,
                    },
                  },
                  "soundcloud": Object {
                    "options": Object {
                      "buying": false,
                      "download": false,
                      "liking": false,
                      "sharing": false,
                      "show_comments": false,
                      "show_playcount": false,
                      "visual": true,
                    },
                  },
                  "tracks": Array [],
                  "twitch": Object {
                    "options": Object {},
                    "playerId": null,
                  },
                  "vidyard": Object {
                    "options": Object {},
                  },
                  "vimeo": Object {
                    "playerOptions": Object {
                      "autopause": false,
                      "byline": false,
                      "portrait": false,
                      "title": false,
                    },
                  },
                  "wistia": Object {
                    "options": Object {},
                    "playerId": null,
                  },
                  "youtube": Object {
                    "embedOptions": Object {},
                    "onUnstarted": [Function],
                    "playerVars": Object {
                      "iv_load_policy": 3,
                      "modestbranding": 1,
                      "playsinline": 1,
                      "rel": 0,
                      "showinfo": 0,
                    },
                  },
                }
              }
              controls={false}
              height="100%"
              key="file"
              light={false}
              loop={true}
              muted={true}
              onBuffer={[Function]}
              onBufferEnd={[Function]}
              onDisablePIP={[Function]}
              onDuration={[Function]}
              onEnablePIP={[Function]}
              onEnded={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onProgress={[Function]}
              onReady={[Function]}
              onSeek={[Function]}
              onStart={[Function]}
              pip={false}
              playbackRate={1}
              playing={true}
              playsinline={false}
              progressInterval={1000}
              stopOnUnmount={true}
              style={Object {}}
              url={
                Array [
                  Object {
                    "src": "/video/w_1300_blend_mode_multiply.mp4",
                    "type": "video/mp4",
                  },
                ]
              }
              volume={null}
              width="100%"
              wrapper="div"
            />
          </Suspense>
        </div>
      </ReactPlayer>
    </div>
  </styles__Video>
</Memo(WAnimation)>
`;

exports[`<WAnimation /> wrapperWithoutBlendMode: should render without blend mode 1`] = `
<Memo(WAnimation)
  isMixMultiplyBlendMode={false}
>
  <styles__Video
    className="w-animation"
    isMixMultiplyBlendMode={false}
  >
    <div
      className="styles__Video-sc-1iarrbt-0 foMWSv w-animation"
    >
      <ReactPlayer
        autoPlay={true}
        className="react-player"
        config={
          Object {
            "dailymotion": Object {
              "params": Object {
                "api": 1,
                "endscreen-enable": false,
              },
            },
            "facebook": Object {
              "appId": "1309697205772819",
              "attributes": Object {},
              "playerId": null,
              "version": "v3.3",
            },
            "file": Object {
              "attributes": Object {},
              "dashVersion": "3.1.3",
              "flvVersion": "1.5.0",
              "forceAudio": false,
              "forceDASH": false,
              "forceFLV": false,
              "forceHLS": false,
              "forceVideo": false,
              "hlsOptions": Object {},
              "hlsVersion": "0.14.16",
              "tracks": Array [],
            },
            "mixcloud": Object {
              "options": Object {
                "hide_cover": 1,
              },
            },
            "soundcloud": Object {
              "options": Object {
                "buying": false,
                "download": false,
                "liking": false,
                "sharing": false,
                "show_comments": false,
                "show_playcount": false,
                "visual": true,
              },
            },
            "twitch": Object {
              "options": Object {},
              "playerId": null,
            },
            "vidyard": Object {
              "options": Object {},
            },
            "vimeo": Object {
              "playerOptions": Object {
                "autopause": false,
                "byline": false,
                "portrait": false,
                "title": false,
              },
            },
            "wistia": Object {
              "options": Object {},
              "playerId": null,
            },
            "youtube": Object {
              "embedOptions": Object {},
              "onUnstarted": [Function],
              "playerVars": Object {
                "iv_load_policy": 3,
                "modestbranding": 1,
                "playsinline": 1,
                "rel": 0,
                "showinfo": 0,
              },
            },
          }
        }
        controls={false}
        height="100%"
        light={false}
        loop={true}
        muted={true}
        onBuffer={[Function]}
        onBufferEnd={[Function]}
        onDisablePIP={[Function]}
        onDuration={[Function]}
        onEnablePIP={[Function]}
        onEnded={[Function]}
        onError={[Function]}
        onPause={[Function]}
        onPlay={[Function]}
        onProgress={[Function]}
        onReady={[Function]}
        onSeek={[Function]}
        onStart={[Function]}
        pip={false}
        playbackRate={1}
        playing={true}
        playsinline={false}
        progressInterval={1000}
        stopOnUnmount={true}
        style={Object {}}
        url={
          Array [
            Object {
              "src": "/video/w_2048.mp4",
              "type": "video/mp4",
            },
          ]
        }
        volume={null}
        width="100%"
        wrapper="div"
      >
        <div
          autoPlay={true}
          className="react-player"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <Suspense
            fallback={null}
          >
            <Player
              activePlayer={
                Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": [Function],
                  "_status": 1,
                }
              }
              autoPlay={true}
              className="react-player"
              config={
                Object {
                  "attributes": Object {},
                  "dailymotion": Object {
                    "params": Object {
                      "api": 1,
                      "endscreen-enable": false,
                    },
                  },
                  "dashVersion": "3.1.3",
                  "facebook": Object {
                    "appId": "1309697205772819",
                    "attributes": Object {},
                    "playerId": null,
                    "version": "v3.3",
                  },
                  "file": Object {
                    "attributes": Object {},
                    "dashVersion": "3.1.3",
                    "flvVersion": "1.5.0",
                    "forceAudio": false,
                    "forceDASH": false,
                    "forceFLV": false,
                    "forceHLS": false,
                    "forceVideo": false,
                    "hlsOptions": Object {},
                    "hlsVersion": "0.14.16",
                    "tracks": Array [],
                  },
                  "flvVersion": "1.5.0",
                  "forceAudio": false,
                  "forceDASH": false,
                  "forceFLV": false,
                  "forceHLS": false,
                  "forceVideo": false,
                  "hlsOptions": Object {},
                  "hlsVersion": "0.14.16",
                  "mixcloud": Object {
                    "options": Object {
                      "hide_cover": 1,
                    },
                  },
                  "soundcloud": Object {
                    "options": Object {
                      "buying": false,
                      "download": false,
                      "liking": false,
                      "sharing": false,
                      "show_comments": false,
                      "show_playcount": false,
                      "visual": true,
                    },
                  },
                  "tracks": Array [],
                  "twitch": Object {
                    "options": Object {},
                    "playerId": null,
                  },
                  "vidyard": Object {
                    "options": Object {},
                  },
                  "vimeo": Object {
                    "playerOptions": Object {
                      "autopause": false,
                      "byline": false,
                      "portrait": false,
                      "title": false,
                    },
                  },
                  "wistia": Object {
                    "options": Object {},
                    "playerId": null,
                  },
                  "youtube": Object {
                    "embedOptions": Object {},
                    "onUnstarted": [Function],
                    "playerVars": Object {
                      "iv_load_policy": 3,
                      "modestbranding": 1,
                      "playsinline": 1,
                      "rel": 0,
                      "showinfo": 0,
                    },
                  },
                }
              }
              controls={false}
              height="100%"
              key="file"
              light={false}
              loop={true}
              muted={true}
              onBuffer={[Function]}
              onBufferEnd={[Function]}
              onDisablePIP={[Function]}
              onDuration={[Function]}
              onEnablePIP={[Function]}
              onEnded={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onProgress={[Function]}
              onReady={[Function]}
              onSeek={[Function]}
              onStart={[Function]}
              pip={false}
              playbackRate={1}
              playing={true}
              playsinline={false}
              progressInterval={1000}
              stopOnUnmount={true}
              style={Object {}}
              url={
                Array [
                  Object {
                    "src": "/video/w_2048.mp4",
                    "type": "video/mp4",
                  },
                ]
              }
              volume={null}
              width="100%"
              wrapper="div"
            />
          </Suspense>
        </div>
      </ReactPlayer>
    </div>
  </styles__Video>
</Memo(WAnimation)>
`;
